jlm/llvm/backend/RvsdgToIpGraphConverter.cpp

```diff
@@ -277,25 +278,58 @@ RvsdgToIpGraphConverter::ConvertEmptyGammaNode(const rvsdg::GammaNode & gammaNod
       continue;
     }
 
-    const auto node = rvsdg::output::GetNode(*predicate);
-    if (is<rvsdg::match_op>(node))
+    const auto matchNode = rvsdg::output::GetNode(*gammaPredicate);
+    if (is<rvsdg::match_op>(matchNode))
     {
-      const auto matchOperation = util::AssertedCast<const rvsdg::match_op>(&node->GetOperation());
-      const auto defaultAlternative = matchOperation->default_alternative();
-      const auto condition = Context_->GetVariable(node->input(0)->origin());
-      const auto trueAlternative =
-          defaultAlternative == 0 ? Context_->GetVariable(output1) : Context_->GetVariable(output0);
-      const auto falseAlternative =
-          defaultAlternative == 0 ? Context_->GetVariable(output0) : Context_->GetVariable(output1);
+      const auto matchOperation =
+          util::AssertedCast<const rvsdg::match_op>(&matchNode->GetOperation());
+      assert(matchOperation->nalternatives() == 2);
+
+      const auto matchOrigin = Context_->GetVariable(matchNode->input(0)->origin());
+      const auto caseValue = matchOperation->begin()->first;
+      const auto caseSubregion = matchOperation->begin()->second;
```

Comment:
At this point we have asserted that the match only has two alternatives, but does that necessarily mean that it only has one (value->alt) mapping, and one default alternative? If it has multiple mapping then we might not be able to convert it into a select, but has that case been determined to never occur before reaching this point in the code?

Reply:
It should, but the input from the match operation is not really well checked (to put it nicely). We should improve this. I put another `JLM_ASSERT()` there that ensures that we really have only two cases. At least, it will scream at us when this is not the case.